src/Sulu/Component/SmartContent/Configuration/ProviderConfigurationInterface.php

```diff
@@ -52,6 +52,18 @@ public function setAudienceTargeting(bool $audienceTargeting);
      */
     public function hasTags(): bool;
 
+    /**
+     * Returns types for types filtering.
+     *
+     * @return null|PropertyParameter[]
+     */
+    public function getTypes(): ?array;
+
+    /**
+     * Returns TRUE if types should be displayed.
+     */
+    public function hasTypes(): bool;
```

Comment:
Same as above, strictly speaking that would be a BC break, but it is not necessary to implement this interface to add your own `DataProvider`, so I think it should be fine.